code/modules/reagents/chemistry/reagents/toxin_reagents.dm

```diff
@@ -106,6 +106,12 @@
 	if(holder.has_reagent(/datum/reagent/medicine/epinephrine))
 		holder.remove_reagent(/datum/reagent/medicine/epinephrine, 2 * REM * delta_time)
 	affected_mob.adjustPlasma(20 * REM * delta_time)
+	var/obj/item/organ/internal/liver/liver = affected_mob.getorganslot(ORGAN_SLOT_LIVER)
+	if(liver && HAS_TRAIT(liver, TRAIT_PLASMA_LOVER_METABOLISM)) // Some livers can metabolize plasma
+		toxpwr = 0
+		if(isplasmaman(affected_mob)) // having wounds repaired still requires plasmaman biology, in case of transplanted plasma livers
+			for(var/datum/wound/iter_wound as anything in affected_mob.all_wounds)
+				iter_wound.on_xadone(4 * REAGENTS_EFFECT_MULTIPLIER * delta_time) // plasmamen use plasma to reform their bones or whatever
```

Comment:
```
/datum/reagent/toxin/plasma/on_mob_metabolize(mob/living/carbon/affected_mob)
	var/obj/item/organ/internal/liver/liver = affected_mob.getorganslot(ORGAN_SLOT_LIVER)
	if(liver && HAS_TRAIT(liver, TRAIT_PLASMA_LOVER_METABOLISM)) // Some livers can metabolize plasma
		toxpwr = 0

/datum/reagent/toxin/plasma/on_mob_end_metabolize(mob/living/carbon/affected_mob)
	toxpwr = initial(toxpwr)

/datum/reagent/toxin/plasma/on_mob_life(mob/living/carbon/affected_mob, delta_time, times_fired)
	if(holder.has_reagent(/datum/reagent/medicine/epinephrine))
		holder.remove_reagent(/datum/reagent/medicine/epinephrine, 2 * REM * delta_time)
	affected_mob.adjustPlasma(20 * REM * delta_time)
	if(isplasmaman(affected_mob)) // having wounds repaired still requires plasmaman biology, in case of transplanted plasma livers
		for(var/datum/wound/iter_wound as anything in affected_mob.all_wounds)
			iter_wound.on_xadone(4 * REAGENTS_EFFECT_MULTIPLIER * delta_time) // plasmamen use plasma to reform their bones or whatever
	return ..()
```

CC @MrMelbert how does this look?  

- Plasmaman liver determines if they take toxin damage from plasma
- Plasmaman biology heals wounds separately (even without the liver)

So a human with a plasmaman liver, can ignore plasma toxin but without the regen abilities.  A plasmaman with a human liver will take toxin damage, but the rest of his body will heal wounds.  Sound reasonable?

Reply:
One last thing I am confused about-- let's say you removed a plasmaman's liver. I would expect plasmamen to not take tox damage from that and looking at the code, it seems like that was the intent. However, that is not the case from my testing.

In reagents.dm:

`var/affected_biotype = MOB_ORGANIC` Note that the plasmaman species is not `MOB_ORGANIC`, it is `MOB_MINERAL` and `MOB_HUMANOID`

Then we have this:

```
/datum/reagent/toxin/on_mob_life(mob/living/carbon/affected_mob, delta_time, times_fired)
	if(toxpwr && affected_mob.health > health_required)
		affected_mob.adjustToxLoss(toxpwr * REM * normalise_creation_purity() * delta_time, FALSE, required_biotype = affected_biotype)
		. = TRUE
	..()
```
Note the `required_biotype = affected_biotype` bit.

From /code/modules/mob/living/carbon/damage_procs.dm:
```

/mob/living/carbon/adjustToxLoss(amount, updating_health = TRUE, forced = FALSE, required_biotype)
	if(required_biotype && !(mob_biotypes & required_biotype))
		return

```
Given required_biotype is set to `MOB_ORGANIC` and plasmamen have `MOB_MINERAL `
adjustToxLoss should return here, right? With no tox damage applied? Am I missing something?

Reply:
@MrMelbert @timothymtorres After #72664 a thought popped into my head--plasmamen shouldn't be taking organ damage from _any_ toxins that target organics, right? So long as they have their plasmaman liver. And so I came up with a better, more general solution than the one before.

Since we already have affected_biotype why not add a liver_biotype (using the same bitflags as mob_biotype) that matches against that?

By default all of the liver_biotypes are set to MOB_ORGANIC, except plasmamen which are MOB_MINERAL.
This will allow for future additions of species-targeting toxins to not require snowflake code to implement. Now add the relevant biotype and it will just work.

This will make the plasma_liver_metabolism trait go unused--I left it in for now, but can remove it if you'd like. Might be useful somewhere down the line perhaps if you want to specify plasma.